licenses.yaml

```diff
@@ -157,6 +157,15 @@ libraries:
 
 ---
 
+name: Code adopted from org.apache.commons.dbcp2.BasicDataSource
+license_category: source
+module: server
+license_name: Apache License version 2.0
+source_paths:
+  - server/src/main/java/org/apache/druid/metadata/BasicDataSourceExt.java
```

Comment:
Can you please add a notice here too? The notice for commons-dbcp can be found in https://github.com/apache/commons-dbcp/blob/master/NOTICE.txt.

Reply:
updated, thanks for the pointers, really appreciate it as I am somewhat of a noob in my understanding of all contents in license file and just getting by with the most powerful tool there is i.e. the copy-and-paste :)

Reply:
Since it's been long to update this source license, the update process was not clear to me as well. I wrote some in https://github.com/apache/druid/pull/10894. Based on the [instructions in that PR](https://github.com/apache/druid/pull/10894/files#diff-e0657fbb35e792bf869c4bd37b330e5bd8ad37b426296f153839512d25dbcf39R48), you need to update the `LICENSE` and `NOTICE` files for source code packages as well since they are currently managed manually. Also, the CI failure seems legit as the version is missing here.

```
=== Generating the contents of NOTICE.BINARY file ===

/home/travis/build/apache/druid/distribution/bin/generate-binary-notice.py:60: YAMLLoadWarning: calling yaml.load_all() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  dependencies = list(yaml.load_all(registry_file))

Traceback (most recent call last):
  File "/home/travis/build/apache/druid/distribution/bin/generate-binary-notice.py", line 88, in <module>
    generate_notice(source_notice, dependencies_yaml)
  File "/home/travis/build/apache/druid/distribution/bin/generate-binary-notice.py", line 72, in generate_notice
    print_notice(dependency)
  File "/home/travis/build/apache/druid/distribution/bin/generate-binary-notice.py", line 40, in print_notice
    print_outfile("{} {} {} {}".format(dependencyHeaderLine, dependency['name'], dependency['version'], dependencyHeaderLine))
KeyError: 'version'
```

Reply:
thanks for the pointers. I think I have updated all things now.